src/redux/missions/missions.js

```diff
@@ -0,0 +1,62 @@
+const apiEndpoint = 'https://api.spacexdata.com/v3/missions';
+const initialState = [];
+
+const ADD_MISSION = 'missions/ADD_MISSION';
+const JOIN_OR_LEAVE_MISSION = 'missions/JOIN_OR_LEAVE_MISSION';
+
+const fetchAllMissions = async () => {
+  let res = await fetch(apiEndpoint);
+  res = await res.json();
+  return res;
+};
+
+const addMission = (mission) => ({
+  type: ADD_MISSION,
+  payload: mission,
+});
+
+export const toggleMissionState = (mission) => ({
+  type: JOIN_OR_LEAVE_MISSION,
+  payload: mission,
+});
+
+export const fetchMissions = async (dispatch, getState) => {
+  if (getState().missionsReducer.length > 0) return;
+  const missions = await fetchAllMissions();
+  missions.forEach((mission) => {
+    const newMission = {
+      mission_id: mission.mission_id,
+      mission_name: mission.mission_name,
+      description: mission.description,
+    };
+    dispatch(addMission(newMission));
+  });
+};
```

Comment:
- [ ] When you take the missions info at this point, you add every mission one by one user and action, a better way to do it is to add all of them in an object and once you're here on the reducer, add it with another switch option, that would save the memory cost to run an action a lot of times. Please use this on this redux. 🐶